test/integration/index_management.test.js

```diff
@@ -258,39 +271,62 @@ describe('Indexes', function () {
     }
   });
 
-  it('shouldCorrectlyDropIndexes', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
+  context('when dropIndexes succeeds', function () {
+    let collection;
 
-    test: function (done) {
-      var configuration = this.configuration;
-      var client = configuration.newClient(configuration.writeConcernMax(), { maxPoolSize: 1 });
-      var db = client.db(configuration.db);
-      db.createCollection('test_drop_indexes', function (err, collection) {
-        collection.insert({ a: 1 }, configuration.writeConcernMax(), function (err) {
-          expect(err).to.not.exist;
-          // Create an index on the collection
-          db.createIndex(
-            collection.collectionName,
-            'a',
-            configuration.writeConcernMax(),
-            function (err, indexName) {
-              test.equal('a_1', indexName);
-              // Drop all the indexes
-              collection.dropIndexes(function (err, result) {
-                test.equal(true, result);
+    beforeEach(async function () {
+      collection = await db.createCollection('test_drop_indexes');
+      await collection.insert({ a: 1 });
+      // Create an index on the collection
+      await db.createIndex(collection.collectionName, 'a');
+    });
 
-                collection.indexInformation(function (err, result) {
-                  test.ok(result['a_1'] == null);
-                  client.close(done);
-                });
-              });
-            }
-          );
+    afterEach(async function () {
+      await db.dropCollection('test_drop_indexes');
+    });
+
+    it('should return true and should no longer exist in the collection', async function () {
+      // Drop all the indexes
+      const result = await collection.dropIndexes();
+      expect(result).to.equal(true);
+
+      const res = await collection.indexInformation();
+      expect(res['a_1']).to.equal(undefined);
+    });
+  });
+
+  context('when dropIndexes fails', function () {
+    let collection;
+
+    beforeEach(async function () {
+      collection = await db.createCollection('test_drop_indexes');
+      await collection.insert({ a: 1 });
+      // Create an index on the collection
+      await db.createIndex(collection.collectionName, 'a');
+      /**@type {import('../tools/utils').FailPoint} */
+      await client
```

Comment:
it looks like tests are failing here, because failCommands are only available on 4.0+ servers: https://github.com/mongodb/mongo/wiki/The-%22failCommand%22-fail-point

the test in question is testing driver error handling of the server error.  I think it's fine to restrict this test to run only on 4.0+ servers, with a comment explaining why the restriction is there.

We should still test `createIndex` on all servers though.

Reply:
how do you restrict it to specific servers